compiler/compiler.go

```diff
@@ -37,6 +37,7 @@ type Archive struct {
 	Imports      []string
 	ExportData   []byte
 	Declarations []*Decl
+	JSDecls      []*Decl
```

Comment:
I think we could use a simple `[]byte` here instead of `[]*Decl` and rename it, maybe `IncJSCode`?